docs/source/reference-core.rst

```diff
@@ -834,13 +834,13 @@ Nursery objects provide the following interface:
 
       Note that this is *not* an async function and you don't use await
       when calling it. It sets up the new task, but then returns
-      immediately, *before* it has a chance to run. The new task won’t
-      actually get a chance to do anything until some later point when
-      you execute a checkpoint and the scheduler decides to run it.
-      If you want to run a function and immediately wait for its result,
-      then you don't need a nursery; just use ``await async_fn(*args)``.
-      If you want to wait for the task to initialize itself before 
-      continuing, see :meth:`start()`.
+      immediately, *before* it has a chance to run. At some future
+      time, (which might be closing of the nursery context, but no
+      later), all tasks requested with :meth:`start_soon` will begin,
```

Comment:
This still could be read to imply that all of the tasks will be started at once, which isn't a guarantee we want to make. What if anything do you think is unclear about taking the previous wording ("won't actually get a chance to do anything ... the scheduler decides to run it") and adding a sentence like "Tasks might start running in a different order than the calls to `start_soon()`"?

Reply:
You're right - it does imply a single time. Rephrased to "All tasks requested with :meth:`start_soon` will begin, at no particular time, in no particular order (but no later than the closing of the nursery context)."
I wanted to avoid mention of the scheduler here. The important thing is that ordering and timing get arbitrary when you call start_soon - I don't think the reader will get that unless they have a good grasp of what the scheduler does and does not promise.